cli/flox/src/commands/generations/rollback.rs

```diff
@@ -0,0 +1,49 @@
+use anyhow::Result;
+use bpaf::Bpaf;
+use flox_rust_sdk::flox::Flox;
+use flox_rust_sdk::models::environment::generations::{GenerationsEnvironment, GenerationsExt};
+use itertools::Itertools;
+use tracing::{debug, instrument};
+
+use crate::commands::{EnvironmentSelect, environment_select};
+use crate::environment_subcommand_metric;
+use crate::utils::message;
+
+/// Arguments for the `flox generations rollback` command
+#[derive(Bpaf, Debug, Clone)]
+pub struct Rollback {
+    #[bpaf(external(environment_select), fallback(Default::default()))]
+    environment: EnvironmentSelect,
+}
+
+impl Rollback {
+    #[instrument(name = "rollback", skip_all)]
+    pub fn handle(self, flox: Flox) -> Result<()> {
+        let env = self.environment.to_concrete_environment(&flox)?;
+
+        environment_subcommand_metric!("generations::rollback", env);
+        let mut env: GenerationsEnvironment = env.try_into()?;
+
+        debug!("determining previous generation");
+        let metadata = env.generations_metadata()?;
+
+        // (0, is the current active)
+        let Some((previously_active_generation_id, _meta)) = metadata
+            .generations
+            .iter()
+            .sorted_by_key(|(_id, meta)| meta.last_active)
+            .nth(1)
+        else {
+            message::info("No previous generation to rollback to.");
```

Comment:
Blocker: I think this is more of a warning or error situation e.g. "we cannot do what you asked".